bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/handler/AstroThingHandler.java

```diff
@@ -167,8 +167,8 @@ public void publishPlanet() {
      */
     public void publishChannelIfLinked(ChannelUID channelUID) {
         Planet planet = getPlanet();
-        if (isLinked(channelUID.getId()) && planet != null) {
-            final Channel channel = getThing().getChannel(channelUID.getId());
+        if (isLinked(channelUID) && planet != null) {
```

Comment:
The same improvement can be made in line 275.

Reply:
@lsiepel thanks for the hint, i have updated the PR.